runtime/lua/vim/lsp.lua

```diff
@@ -1967,7 +1964,7 @@ end
 --  Unlike `buf_request`, this will collect all the responses from each server instead of handling them.
 --  A map of client_id:request_result will be provided to the callback
 --
----@returns (function) A function that will cancel all requests which is the same as the one returned from `buf_request`.
+---@return fun() # A function that will cancel all requests which is the same as the one returned from `buf_request`.
```

Comment:
The `#` here (for an anonymous return value?) doesn't translate well to vimdoc. Can we find something else?

Reply:
It's to mark the remainder as comment if there is no name:

`---@return <type> [<name> [comment] | [name] #<comment>]`

I have changed it for now and named the return value `cancel` as it fits here. But maybe worth adding some extra logic to the doc generator to strip such a prefix.

Reply:
> But maybe worth adding some extra logic to the doc generator to strip such a prefix.

Yes, that would be best. There's a bunch of other places in the code with a similar problem.